Sources/Testing/Traits/Trait.swift

```diff
@@ -81,6 +81,41 @@ public protocol SuiteTrait: Trait {
   var isRecursive: Bool { get }
 }
 
+/// A protocol extending ``Trait`` that offers an additional customization point
+/// for trait authors to execute code before and after each test function (if
+/// added to the traits of a test function), or before and after each test suite
+/// (if added to the traits of a test suite).
+public protocol CustomExecutionTrait: Trait {
+
+  /// Execute a function with the effects of this trait applied.
+  ///
+  /// - Parameters:
+  ///   - function: The function to perform. If `test` represents a test suite,
+  ///     this function encapsulates running all the tests in that suite. If
+  ///     `test` represents a test function, this function is the body of that
+  ///     test function (including all cases if it is parameterized.)
+  ///   - test: The test under which `function` is being performed.
+  ///   - testCase: The testCase under which `function` is being performed.
+  ///
+  /// - Throws: Whatever is thrown by `function`, or an error preventing the
+  ///   trait from running correctly.
+  ///
+  /// This function is called for each ``CustomExecutionTrait`` on a test suite
+  /// or test function and allows additional work to be performed before or
+  /// after the test runs.
+  /// It gets invoked once for each test, and then once for each test case.
```

Comment:
```suggestion
  /// This function is invoked once for the test it is applied to, and then once for each test case in that test, if applicable.
```

Did you mean for this to be a separate paragraph or a continuation of the previous one?

Reply:
A new paragraph, I inserted another newline.